packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -58,6 +59,36 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
       hour,
       minute,
     };
+  } else if (isWeeklyTriggerInput(userFacingTrigger)) {
+    const weekday = userFacingTrigger.weekday;
+    const hour = userFacingTrigger.hour;
+    const minute = userFacingTrigger.minute;
+    if (
+      weekday === undefined ||
+      weekday == null ||
+      hour === undefined ||
+      hour == null ||
+      minute === undefined ||
+      minute == null
+    ) {
+      throw new TypeError('Weekday, hour and minute need to have valid values. Found undefined');
+    }
+    if (weekday < 1 || weekday > 7) {
+      throw new RangeError(`The weekday parameter needs to be between 1 and 7. Found: ${weekday}`);
```

Comment:
Could we lean API-wise more into what JS Date would return from `getDay`? According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay,

> **Return value**
> An integer number, between 0 and 6, corresponding to the day of the week for the given date, according to local time: 0 for Sunday, 1 for Monday, 2 for Tuesday, and so on.

Having 1 for Sunday is a _native thing_, let's do the change on the native side! 🙏

Reply:
Sure, I have considered this too. But if we take the JS approach (0-6), should I then also take a look at the `CalendarNotificationTrigger`, because it still accepts numbers from 1 through 7?

Reply:
@sjchmiela what do you think about this?

Reply:
Hey, sorry for forgetting to respond. Thank you for noticing this inconsistency! I think we should ship this PR with the API you've written already which is consistent and I'll make another PR afterwards that will change the API in a slightly different, less-thinking-about-native, more-developer-friendly way (hopefully).